### PR TITLE
feat(appengine/queue): Consistent task queue with dependencies.

### PR DESCRIPTION
**Pull request by tartavull**
We need dependecies between tasks.
We have relatively low thoughput of tasks, because each tasks
usually takes serval minutes.

Widely used task queues doesn't support depedencies, so we
decided to reinvet the wheel and write yet another task queue.
This taskqueue, closely follows Google's Pull Queue API.

### CODE REVIEW

**Comment by tartavull** [comment]
I haven't yet deployed, so this is not ready to be reviewed.
But I wanted to show you guys what I had so far.

**Comment by tartavull** [comment]
queue url
https://queue-dot-neuromancer-seung-import.appspot.com/myproject/taskqueue/myqueue/tasks

File: appengine/queue/taskqueue.py [hunk 1]
```diff
@@ -0,0 +1,397 @@
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+import logging
+import webapp2
```
Comment (nkemnitz) [inline appengine/queue/taskqueue.py]: webapp2 is not part of any requirements.txt, I think

Comment (tartavull) [inline appengine/queue/taskqueue.py]: https://cloud.google.com/appengine/docs/standard/python/tools/built-in-libraries-27

I don't think it's needed

File: appengine/queue/taskqueue.py [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+        assert newLeaseSeconds >= 1
+        self._patch_task(queue, task_name, newLeaseSeconds)
+
+    def delete(self, project_name, taskqueue_name, task_name):
+        queue =  queues[taskqueue_name]
+        self._delete_task(queue, task_name)
+
+    def _delete_task(self, queue, task_name):
+        """
+        DELETE  /projectname/taskqueue/queuename/tasks/taskname
```
Comment (nkemnitz) [inline appengine/queue/taskqueue.py]: Can you emphasize the parts of the call the user has to fill in? Maybe `/${projectname}/taskqueue/${queuename}/tasks/${taskname}` or something like that? Same for all the other requests.

**Pull request merged**